pkg/controller/namespace/namespace_controller.go

```diff
@@ -169,10 +169,19 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 	}
 
 	if !controlledByWorkspace {
+		err = r.unbindWorkspace(instance)
```

Comment:
To make it shorter and more clear, I suggest
```suggestion
		if err := r.unbindWorkspace(instance); err != nil {
		   return reconcile.Result{}, err
		}
```